var/spack/repos/builtin/packages/cmake/package.py

```diff
@@ -114,6 +114,23 @@ class Cmake(Package):
         values=("Debug", "Release", "RelWithDebInfo", "MinSizeRel"),
     )
 
+    # We default ownlibs to true because it greatly speeds up the CMake
+    # build, and CMake is built frequently. Also, CMake is almost always
+    # a build dependency, and its libs will not interfere with others in
+    # the build.
+    variant("ownlibs", default=True, description="Use CMake-provided third-party libraries")
+    variant(
+        "doc",
+        default=False,
+        description="Enables the generation of html and man page documentation",
+    )
+    variant(
+        "ncurses",
+        default=sys.platform != "win32",
+        description="Enables the build of the ncurses gui",
+    )
+    variant("qtgui", default=False, description="Enables the build of the Qt GUI")
```

Comment:
@scheibelp @johnwparent, this was removed in https://github.com/spack/spack/pull/39360 for a good reason. Why bring back this circular dependency cmake - qt - cmake? I would suggest to undo this.

Reply:
It was re-introduced because building the Qt GUI is a relevant component of building CMake and we wanted that to be available to users.

Reply:
It is still autoconf package. I agree unsolvable dependecy cycles are bad, so we can use the `build_system` variant to for `cmake +qt` to need `libmng build_system=autoconf` and this should work fine right?

I agree with @johnwparent here that providing the GUI is important, especially on Windows where people expect a GUI.

Reply:
Basic build options using autoconf here: https://www.linuxfromscratch.org/blfs/view/git/general/libmng.html

Reply:
I'll pr the autotools build for libmng, thanks @kwryankrattiger for pointing that out